app/upgrade_cudos.go

```diff
@@ -257,7 +249,26 @@ func getAccPrefix(jsonData map[string]interface{}) (string, error) {
 	return "", fmt.Errorf("failed to get prefix: %w", lastErr)
 }
 
-func parseGenesisData(jsonData map[string]interface{}, genDoc *tmtypes.GenesisDoc, cudosCfg *CudosMergeConfig, manifest *UpgradeManifest) (*GenesisData, error) {
+func getBondDenom(jsonData map[string]interface{}) (string, error) {
+	staking, ok := jsonData["staking"].(map[string]interface{})
+	if !ok {
+		return "", fmt.Errorf("staking module data not found in genesis")
+	}
+
+	stakingParams, ok := staking["params"].(map[string]interface{})
+	if !ok {
+		return "", fmt.Errorf("staking params not found in genesis")
+	}
+
+	bondDenom, ok := stakingParams["bond_denom"].(string)
+	if !ok {
+		return "", fmt.Errorf("staking params bond denom value not found in genesis")
+	}
+
+	return bondDenom, nil
+}
+
+func parseGenesisData(app *App, ctx sdk.Context, jsonData map[string]interface{}, genDoc *tmtypes.GenesisDoc, cudosCfg *CudosMergeConfig, manifest *UpgradeManifest) (*GenesisData, error) {
```

Comment:
What was the motivation behind this changing passing from by-pointer to by-value? Was it attempt to disable theoreticall possibility of accidental modification of `ctx` instance value?

The `sdk.Context` is quite big structure, from performance perspective it would make sense to keep passibg by-pointer.

Reply:
It was just unification with other functions. I can unify it to convert everything to pointer.

Reply:
Actually some CosmoSDK function expects `sdk.Context` instead of `*sdk.Context` for example `GetValidator(ctx sdk.Context`